components/EmailValidator.vue

```diff
@@ -0,0 +1,147 @@
+<template>
+  <div class="d-inline-block w-100">
+    <validating-form
+      :class="{ 'justify-content-around': center, 'd-flex': center }"
+    >
+      <b-form-group
+        :label="label"
+        :label-for="'email-' + id"
+      >
+        <validating-form-input
+          :value="email"
+          type="email"
+          name="email"
+          :size="size"
+          class="email"
+          :validation="$v.email"
+          validation-enabled
+          :validation-messages="{
+            email: 'Please enter a valid email address.'
+          }"
+          :center="center"
+          autocomplete="username email"
+          @input="input"
+          @focus="focus"
+          @blur="blur"
+        />
+      </b-form-group>
+    </validating-form>
+    <div v-if="suggestedDomains && suggestedDomains.length" class="text-info small">
+      Did you mean <b>{{ suggestedDomains[0] }}</b>?
+    </div>
+  </div>
+</template>
+<script>
+import { required, email } from 'vuelidate/lib/validators'
+import { validationMixin } from 'vuelidate'
+import ValidatingForm from '../components/ValidatingForm'
+import ValidatingFormInput from '../components/ValidatingFormInput'
+import validationHelpers from '@/mixins/validationHelpers'
+
+const a = require('axios')
+const axios = a.create({
+  timeout: 300000
+})
+
+export default {
+  components: { ValidatingForm, ValidatingFormInput },
+  mixins: [validationMixin, validationHelpers],
+  props: {
+    email: {
+      type: String,
+      required: false,
+      default: null
+    },
+    valid: {
+      type: Boolean,
+      required: true
+    },
+    center: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+    size: {
+      type: String,
+      required: false,
+      default: 'lg'
+    },
+    id: {
+      type: String,
+      required: false,
+      default: ''
+    },
+    label: {
+      type: String,
+      required: false,
+      default: "What's your email address?"
+    }
+  },
+  data: function() {
+    return {
+      suggestedDomains: [],
+      focused: false
+    }
+  },
+  watch: {
+    async email(newVal) {
```

Comment:
I'm not getting any suggesteddomains.  I haven't looked into why yet though.

Reply:
You need to use a domain which is quite similar to a real one, e.g. btintrnet.com

Reply:
Yeah I'm still not getting any.  Might be me doing something stupid though.  I'm just gonna grab some lunch and then I'll take a deeper look.

Reply:
I'm not getting the watch on the email prop running.  It is set once when the component is created but then when I type it's not run again.  Should it be running as I type on the give/whoami page for example?

Reply:
You're right.  The watch works in the login modal, though.  Will dig some more.

Reply:
I've pushed a fix without understanding the problem.  The case where it wasn't working was where we had a prop in the parent which was a computed property.  The set() method of that was firing, but the watch wasn't triggering.

I moved the logic out of a watch and it seems to work better.  But please test.

Reply:
I'll give it a go

Reply:
Now on the give page I'm not actually getting the "Freegle it" button until I change the existing email address.  Does that make sense?  I'll explain better if not...

Reply:
And then when I delete the last character of my email (jayyaj@hotmail.com) and it it back it again then I get the freegle button but it also has a suggestion "do you mean hotmail.com" which is what I've typed

Reply:
I get this on first entry to the page:
![image](https://user-images.githubusercontent.com/17859423/82910843-06a0c680-9f63-11ea-89fb-8b2467212409.png)

If I click into the email field and then out again then I get this:

![image](https://user-images.githubusercontent.com/17859423/82910931-2637ef00-9f63-11ea-8346-462a63560154.png)

If I click into the email field and change something and then I get this:
![image](https://user-images.githubusercontent.com/17859423/82910996-3e0f7300-9f63-11ea-8ae0-09648d30fc6d.png)

Reply:
Not sure if that helps

Reply:
Is it because "valid" is only being set "if (email !== this.email)" in the checkState function?